controllers/front/webhook.php

```diff
@@ -160,54 +160,8 @@ public function processTransaction($transaction)
             }
         }
 
-        if (!isset($apiPayment)) {
-            $apiPayments = array();
-            /** @var MollieOrderAlias $transaction */
-            foreach ($transaction->_embedded->payments as $embeddedPayment) {
-                $apiPayment = ResourceFactory::createFromApiResult($embeddedPayment, new Payment($this->module->api));
-                $apiPayments[] = $apiPayment;
-                unset($apiPayment);
-            }
-            if (count($apiPayments) === 1) {
-                $apiPayment = $apiPayments[0];
-            } else {
-                // In case of multiple payments, the one with the paid status is leading
-                foreach ($apiPayments as $payment) {
-                    if (in_array($payment->status, array(PaymentStatus::STATUS_PAID, PaymentStatus::STATUS_AUTHORIZED))) {
-                        $apiPayment = $payment;
-                        break;
-                    }
-                }
-
-                // No paid/authorized payments found, looking for payments with a final status
-                if (!isset($apiPayment)) {
-                    foreach ($apiPayments as $payment) {
-                        if (in_array($payment->status, array(
-                            PaymentStatus::STATUS_CANCELED,
-                            PaymentStatus::STATUS_FAILED,
-                            PaymentStatus::STATUS_EXPIRED,
-                        ))) {
-                            $apiPayment = $payment;
-                            break;
-                        }
-                    }
-                }
-
-                // In case there is no final payments, we are going to look for any pending payments
-                if (!isset($apiPayment)) {
-                    foreach ($apiPayments as $payment) {
-                        if (in_array($payment->status, array(
-                            PaymentStatus::STATUS_PENDING,
-                        ))) {
-                            $apiPayment = $payment;
-                            break;
-                        }
-                    }
-                }
-            }
-            if (isset($apiPayment)) {
-                $apiPayment->metadata = $transaction->metadata;
-            }
+        if (!empty($transaction->id) && Tools::substr($transaction->id, 0, 3) === 'ord') {
+            $apiPayment = $this->module->api->orders->get($transaction->id, array('embed' => 'payments'));
```

Comment:
what is this? Totally unclear what you want to achieve - create an utility which explains this logic  - also a perfect candidate for unit tests.

Reply:
Added utility for this part.